### PR TITLE
Adding OpInfo for `logical_or`, `logical_and`, `logical_xor`

### PR DESCRIPTION
**Pull request by ankitaS11**
This PR addresses #54261. 

This adds OpInfos for binary logical element wise operators. This is my first PR in OpInfos to PyTorch, looking forward to suggestions and any feedback.

cc: @mruberry @krshrimali

### CODE REVIEW

**Review by krshrimali — APPROVED** [review]
Hi, @ankitaS11 - Thanks for the contribution. This looks good to me. I've left a minor comment, on using `unittest.expectedFailure` instead of `unittest.skip`, but other than that - it LGTM. Let's wait for @mruberry's review on this :)

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -1198,6 +1198,23 @@ def sample_inputs_linalg_matrix_power(op_info, device, dtype, requires_grad):
 
     return inputs
 
+def sample_inputs_logical_binaryops(op_info, device, dtype, requires_grad, **kwargs):
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:1201]: This is interesting. What's your thinking on creating a new function to provide sample inputs for these operations vs reusing `sample_inputs_binary_pwise`? 

https://github.com/pytorch/pytorch/blob/b606df2beea388190dbaeb8d51271bdc30d13a7d/torch/testing/_internal/common_methods_invocations.py#L1652

**Comment by mruberry** [comment]
Hi @ankitaS11! Thank you so much for submitting this PR adding OpInfos for these elementwise binary logical operators. Welcome to the community of contributors. And thank you, @krshrimali, for helping with the review.

The PR looks very good and the one test failure is unrelated (sorry about that!), so don't worry about it.

I have one question about whether this PR should reuse an existing elementwise binary sample inputs function or it needs to create its own. Looking forward to hearing your thoughts!

File: torch/testing/_internal/common_methods_invocations.py (hunk 1 above)
Comment (ankitaS11) [reply torch/testing/_internal/common_methods_invocations.py:1201]: Thanks @mruberry, that's a good catch! I was not aware of `sample_inputs_binary_pwise` function.  I will be using this in my next commit. Looks like it covers all the test cases that are used in `logical_or`, `logicaal_and`, `logical_xor`.

**Review by mruberry — APPROVED** [review]
Cool; thanks @ankitaS11!

**Pull request closed without merging**